File: solutions/csharp/rna-transcription/1/RnaTranscription.cs
using System.Collections.Generic;
using System.Text;

public static class RnaTranscription
{
    private static Dictionary<char, char> strand = new()
    {
        {'G', 'C'},{'C', 'G'},{'T', 'A'},{'A', 'U'}
    };
    public static string ToRna(string nucleotide)
    {
        StringBuilder stringBuilder = new();
        foreach (char ch in nucleotide)
        {
            if (strand.ContainsKey(ch))
                stringBuilder.Append(strand[ch]);
        }
        return stringBuilder.ToString();
    }
}

File: solutions/cpp/rna-transcription/1/rna_transcription.cpp
#include "rna_transcription.h"
#include <stdexcept>
#include <string>

namespace rna_transcription {
char to_rna(char dna) {
  switch (dna) {
  case 'G':
    return 'C';
  case 'C':
    return 'G';
  case 'T':
    return 'A';
  case 'A':
    return 'U';
  default:
    std::string msg = "'" + std::to_string(dna) + "'" + " is not valid.";
    throw std::invalid_argument(msg);
  }
}
std::string to_rna(const std::string &dna) {
  std::string result{};
  for (char ch : dna) {
    result.push_back(to_rna(ch));
  }
  return result;
}

} // namespace rna_transcription
